### ISSUE TITLE
Logging subsystem doesn't work with prink char_out functions

### ISSUE THREAD

**Issue by andyross** [zephyrproject-rtos/zephyr#11975]
If I have a platform where printk() works to emit to a console, the logging system should work with that output stream by default.

Right now, logging only understands a handful of backends, and oddly the printk char_out handler (c.f. __printk_hook_install()) is not one of them.

Obviously on many platforms the printk output is a uart driver, which is a supported backend.  But not all platforms work that way.  Xtensa on qemu has a virtio style console device, which logging doesn't know about (i.e. we have a bunch of tests whitelisted on xtensa simply because they require LOG=y!).  Likewise x86_64 has a default output handler that can emit to the VGA console, which is not a serial backend for logging.

Basically: we already had a "lowest level console output primitive".  Logging needs to use that as its default and not higher level driver abstractions.

**Comment by nordic-krch** [comment]
@andyross since `log_backend_xtensa_sim` was added, could that one be closed?

**Comment by andyross** [comment]
We'll just have the same problem on the next new platform.  The point is that there's an existing "lowest level" output primitive already that we can't really deprecate, and the logging subsystem doesn't support it.

**Comment by andyross** [comment]
*(Doing a grooming pass on bugs/issues that are sitting around with my name on or near them.  No one pays me right now so I get to be an unaccountable whirlwind of chaos!)*

Closing.  In fact LOG_MINIMAL now more or less fills this need.  The twisty knot of output backends is still there, but the hole I was originally whining about is plugged.
